.github/actions/sdk-codegen/publish.sh

```diff
@@ -0,0 +1,111 @@
+#!/usr/bin/env bash
+set -e
+
+# Don't continue if there are no files in the /repo directory
+if [ -z "$(ls -A ./)" ]; then
+  echo "Error: directory is empty"
+  exit 1
+fi
+
+# Don't continue if there were no changes
+if git diff-index --quiet HEAD --; then
+  echo "No changes after code generation, stopping"
+  exit 0
+fi
+
+# ============================================================
+# > SET UP GIT
+# ------------------------------------------------------------
+#
+# GitHub credentials require only a `GITHUB_TOKEN` environment
+# variable, but `git` credentials require explicit setup.
+#
+# ============================================================
+
+git config credential.helper '!f() { sleep 1; echo "username=${GITHUB_USER}"; echo "password=${GITHUB_TOKEN}"; }; f'
```

Comment:
Looks like you could also use the github cli for this:
https://cli.github.com/manual/gh_auth_login
```
gh auth login --with-token < ${GITHUB_TOKEN}
```

Reply:
Yeah, that seems simpler than what we have here.